### ISSUE TITLE
cron identifier not added (not supported?) when 'special' used

### ISSUE THREAD

**Issue by swalladge** [saltstack/salt#38425]
### Description of Issue/Question

When using the 'special' key to give a time period like `@hourly`, the identifier comment isn't added above the cronjob in the crontab. This means that if the command is changed, a new cronjob is added instead of the correct one modified.

### Setup

testfile.sls

```
test cronjob:
  cron.present:
    - name: 'echo foo'
    - special: '@hourly'
    - identifier: some cronjob
```

### Steps to Reproduce Issue

1. run the file on a minion
2. edit 'echo foo' to 'echo bar'
3. rerun the file on the same minion
4. view the crontab on the minion

Expected something like the following:

```
# SALT_CRON_IDENTIFIER:test cronjob
@hourly echo bar
````

Instead I find:

```
@hourly echo foor
@hourly echo bar
```

### Versions Report

Salt Version:
           Salt: 2016.11.1
 
Dependency Versions:
           cffi: 1.9.1
       cherrypy: 8.1.2
       dateutil: 2.6.0
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.24.0
           Mako: 1.0.6
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov  7 2016, 11:55:55)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.4.2
            ZMQ: 4.2.0
 
System Versions:
           dist:   
        machine: x86_64
        release: 4.8.13-1-ARCH
         system: Linux
        version: Not Installed

**Comment by swalladge** [comment]
Note: I also discoved in the docs that the cron module has a `set_job` and `set_special` function, where `set_special` doesn't support an `identifier` argument. Perhaps this is related and also not supported when using `cron.present` in a state.

Relevant pages in the docs:

- [salt.states.cron](https://docs.saltstack.com/en/latest/ref/states/all/salt.states.cron.html#salt.states.cron.present)
- [salt.modules.cron](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.cron.html#salt.modules.cron.set_special)

**Comment by gtmanfred** [comment]
Thanks for the pull request to fix this!

Daniel

**Comment by cebe** [comment]
As far as I see this is not fixed and still relevant. Please reopen.

**Comment by swalladge** [comment]
@cebe if this is a blocker for you, a possible workaround could be to track files under `/etc/cron.d/` instead of managing crontab files with salt.

**Comment by cebe** [comment]
@stale still relevant.

**Comment by cebe** [comment]
Still relevant on salt-ssh 3004.

The issue applies to `cron.present` as well as `cron.absent`. Cron entries are not removed.

**Comment by dwoz** [comment]
Cannot reproduce on 3006.x/3007.x/3008.x/master. Identifier comment is written and re-running with a changed command updates the entry in place. Regression coverage landed in c89a0c19f02 (test_present_special, test_present_special_after_unspecial, test_present_unspecial_after_special) and is present on every supported branch.
